### PR TITLE
Add function to verify format date

### PR DESCRIPTION
**Pull request by jcalarcon98**

### CODE REVIEW

**Review by jeffqev — APPROVED** [review]

File: src/utils/date_utils.py [hunk 1]
```diff
@@ -20,6 +20,22 @@ def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date
 def is_current_date(date: str, date_format: str = '%Y-%m-%d') -> bool:
     if not date:
         return False
+
+    if not is_valid_format_date(date, date_format):
+        return False
+
```
Comment (Wason1797) [inline src/utils/date_utils.py]: Maybe combine both validations in the same if like `if not date or not is_valid_format_date() return False`

And maybe rename the function to -> `is_valid_date_format()`

File: src/utils/date_utils.py [hunk 2]
```diff
@@ -20,6 +20,22 @@ def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date
...
     current_day_month = get_current_day_month(EnvManager.UTC_HOUR_OFFSET)
     date = datetime.strptime(date, date_format)
     return current_day_month == (date.day, date.month)
+
+
+def is_valid_format_date(date: str, date_format: str) -> bool:
```
Comment (Wason1797) [inline src/utils/date_utils.py]: maybe rename this to `is_valid_date_format()`

File: src/utils/date_utils.py [hunk 3]
```diff
@@ -20,6 +20,22 @@ def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date
...
+    if not date:
+        return False
+
+    try:
+        datetime.strptime(date, date_format)
```
Comment (Wason1797) [inline src/utils/date_utils.py]: Maybe try to validate this in a different way `strptime` is kind of expensive and if we are not doing useful things with it is better to not overuse the function

**Review by Wason1797 — CHANGES_REQUESTED** [review]
Some minor things to address

**Review by Wason1797 — APPROVED** [review]
Looks good !

**Pull request merged**